modules/generator/config.go

```diff
@@ -40,6 +48,13 @@ type Config struct {
 	QueryTimeout          time.Duration `yaml:"query_timeout"`
 	OverrideRingKey       string        `yaml:"override_ring_key"`
 
+	// JoinRing controls whether to join the metrics-generator ring and become
+	// available to serve requests.
+	JoinRing bool `yaml:"join_ring"`
```

Comment:
I feel like `JoinRing` doesn't fully cover what this toggle does. Maybe it's more something like `HostLocalBlocks` or `RunLocalBlocks`?

Reply:
You're right, I think what you're suggesting would be a better name. Lets wait to see what others think and then I'll rename according to the consensus.